torch/testing/_internal/distributed/nn/api/remote_module_test.py

```diff
@@ -0,0 +1,130 @@
+import enum
+from typing import Tuple
+
+import torch
+import torch.testing._internal.dist_utils as dist_utils
+from torch import Tensor, nn
+from torch.distributed.nn.api.remote_module import RemoteModule, _gen_global_unique_name
+from torch.testing._internal.distributed.rpc.rpc_agent_test_fixture import (
+    RpcAgentTestFixture,
+)
+
+
+class ModuleCreationMode(enum.Enum):
+    SCRIPT_MODULE_CREATOR = "script_module_creator"
+    MODULE_CTOR_WITH_INTERFACE = "module_ctor_with_interface"
+    MODULE_CTOR = "module_ctor"
+
+
+@torch.jit.interface
+class MyModuleInterface:
+    def forward(
+        self, tensor: Tensor, number: int, word: str = "default"
+    ) -> Tuple[str, int, Tensor]:
+        pass
+
+
+class MyModule(nn.Module):
+    def __init__(self, first_arg, first_kwarg=-1):
+        super().__init__()
+
+    def forward(
+        self, tensor: Tensor, number: int, word: str = "default"
+    ) -> Tuple[str, int, Tensor]:
+        return word, number, tensor
+
+
+def create_module(first_arg, first_kwarg=-1) -> MyModuleInterface:
+    return MyModule(first_arg, first_kwarg=first_kwarg)
```

Comment:
Does this automatically create a ScriptModule due to the return annotation? Or do we need to do `torch.jit.script(MyModule(..))`.

If this already returns a ScriptModule, should we also have a test where returning `torch.jit.script(MyModule(...))` works as well?

Reply:
We still have to use `torch.jit.script(..)` to transform `MyModule `, which is a `torch.nn.Module`, into a ScriptModule, which is in Python type `torch.jit.RecursiveScriptModule`.